lib/logstorage/block_stream_merger.go

```diff
@@ -133,6 +142,12 @@ func (bsm *blockStreamMerger) mustWriteBlock(bd *blockData, bsw *blockStreamWrit
 			// Slow path - copy the bd to the curr bd.
 			bsm.bd.copyFrom(bd)
 		}
+	case !bsm.rows.hasCapacityFor(bd):
```

Comment:
It looks like, this check must be unified with `if bd.uncompressedSizeBytes >= maxUncompressedBlockSize`.

Reply:
`hasCapacityFor` checks against number of unique fields, but not block size.
This check is required in order to ensure that resulting block will not trigger a panic because total number of unique fields in block cannot exceed 10k.
With fields which are taking low amount of space(e.g. short name and value) it is possible that it will be very well compressed and number of unique fields in block will exceed 10k.